src/openni2_driver.cpp

```diff
@@ -312,18 +312,56 @@ void OpenNI2Driver::applyConfigToOpenNIDevice()
     ROS_ERROR("Could not set auto white balance. Reason: %s", exception.what());
   }
 
-  try
+
+  // Workaound for https://github.com/ros-drivers/openni2_camera/issues/51
+  // This is only needed when any of the 3 setting change.  For simplicity
+  // this check is always performed and exposure set.
+  if( (!auto_exposure_ && !auto_white_balance_) && exposure_ != 0 )
   {
-    if (!config_init_ || (old_config_.exposure != exposure_))
-      device_->setExposure(exposure_);
+    ROS_INFO_STREAM("Forcing exposure set, when auto exposure/white balance disabled");
+    forceSetExposure();
   }
-  catch (const OpenNI2Exception& exception)
+  else
   {
-    ROS_ERROR("Could not set exposure. Reason: %s", exception.what());
+    // Setting the exposure the old way, although this should not have an effect
+    try
+    {
+      if (!config_init_ || (old_config_.exposure != exposure_))
+        device_->setExposure(exposure_);
+    }
+    catch (const OpenNI2Exception& exception)
+    {
+      ROS_ERROR("Could not set exposure. Reason: %s", exception.what());
+    }
   }
 
   device_->setUseDeviceTimer(use_device_time_);
+}
 
+
+
+void OpenNI2Driver::forceSetExposure()
+{
+  int current_exposure_ = device_->getExposure();
+  try
+  {
+    if( current_exposure_ == exposure_ )
+    {
+      if( exposure_ < 254 )
```

Comment:
Just curious; what is this `if` section for?

Reply:
Yeah...that definitely needs some explanation.  The camera will ignore any settings it thinks are the same.  However, sometimes the exposure is wrong (multiple reasons).  In order to "force" the exposure, we increment it (or decrement it if its 255, 255 is the max).  After that, we set it to the desired exposure.